pylint/checkers/non_ascii_names.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) 2021-2022 Carli Freudenberg <carli.freudenberg@energymeteo.de>
+
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+"""All alphanumeric unicode character are allowed in Python but could do
+to similarities in their look, they can be confused.
+
+See: https://www.python.org/dev/peps/pep-0672/#confusable-characters-in-identifiers
+
+The following checkers are intended to make users are aware of these issues.
+"""
+
+import re
+import sys
+from typing import Iterable, List, Optional, Union
+
+from astroid import nodes
+
+import pylint.checkers
+from pylint import interfaces
+from pylint.constants import HUMAN_READABLE_TYPES
+from pylint.lint import PyLinter
+
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
+
+
+if sys.version_info[:2] >= (3, 7):
+    # pylint: disable-next=fixme
+    # TODO: Remove after 3.6 has been deprecated
+    Py37Str = str
+else:
+
+    class Py37Str(str):
+        # Allow Python 3.6 compatibility
+        def isascii(self: str) -> bool:
+            return all("\u0000" <= x <= "\u007F" for x in self)
+
+
+class _AsciiOnlyCheckedNode(Protocol):
+    _is_ascii_only: bool
+
+
+class NonAsciiNamesChecker(pylint.checkers.BaseChecker):
+    """A strict name checker only allowing ASCII
+
+    If your programming guideline defines that you are programming in English,
+    then there should be no need for non ASCII characters in Python Names.
+    Everybody else can simply not use this check.
+
+    Note: This check only checks Names, so it ignores the content of
+          docstrings and comments!
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+    priority = -1
+
+    msgs = {
+        "C0144": (
```

Comment:
This should have its own prefix as it is its own file. You can look at how `missing-param-doc` does this: it sets the previous code + `old-missing-param-doc` as the old-name and then takes the new code + `missing-param-doc` as the new name.

Reply:
I did this on purpose to be backward compatible. Changing this, would mean that ppl that ignored this code suddenly get errors.
Should we really do this?

Reply:
That's what the old-names system is for 😉 

By adding the previous code as the code of the `old-name` but using the previous name as the new name nothing will change for end-users 😄

Reply:
Sorry didn't get this reading it first. Got it now.

Reply:
Fixed in #5643